core/update.php

```diff
@@ -189,36 +189,36 @@ function update_helpful_links() {
  * Displays results of the update script with any accompanying errors.
  */
 function update_results_page() {
-  backdrop_set_title('Backdrop site update');
+  backdrop_set_title(t('Backdrop site update'));
 
   update_task_list();
   // Report end result.
   if (module_exists('dblog') && user_access('access site reports')) {
-    $log_message = 'All errors have been <a href="' . base_path() . '?q=admin/reports/dblog">logged</a>.';
+    $log_message = t('All errors have been <a href="!url">logged</a>.', array('!url' => base_path() . '?q=admin/reports/dblog'));
   }
   else {
-    $log_message = 'All errors have been logged.';
+    $log_message = t('All errors have been logged.');
   }
 
   $output = '';
   if (!isset($_SESSION['update_success'])) {
-    $output = '<p>No updates needed.</p>';
+    $output = '<p>' . t('No updates needed') . '</p>';
   }
   elseif ($_SESSION['update_success']) {
-    $output = '<p>Updates were attempted. If you see no failures below, you may proceed happily back to your <a href="' . base_path() . '">site</a>. Otherwise, you may need to update your database manually.' . ' ' . $log_message . '</p>';
+    $output = '<p>' . t('Updates were attempted. If you see no failures below, you may proceed happily back to your <a href="!url">site</a>. Otherwise, you may need to update your database manually.', array('!url' => base_path())) . ' ' . $log_message . '</p>';
   }
   else {
     $updates_remaining = reset($_SESSION['updates_remaining']);
     list($module, $version) = array_pop($updates_remaining);
-    $message = 'The update process was aborted prematurely while running <strong>update #' . $version . ' in ' . $module . '.module</strong>.' . ' ' . $log_message;
+    $message = t('The update process was aborted prematurely while running') . ' <strong>' . t('update #@version in @module.module', array('@version' => $version, '@module' => $module)) . '</strong>.' . ' ' . $log_message;
```

Comment:
The full sentence should be translated (with HTML tags included, if necessary, as here) rather than translating separate fragments of the sentence and concatenating them, because different languages don't use the same order for those fragments. For example, here's English:

The update process was aborted prematurely while running update #1002 in mymodule.module.

But here's German, where the verb comes at the end:

Der Aktualisierungsprozess wurde vorzeitig abgebrochen, während die Aktualisierung Nr. 1002 in der Datei mymodule.module ausgeführt wurde.

So, use this:

```suggestion
    $message = t('The update process was aborted prematurely while running <strong>update #@version in @module.module</strong>.', array('@version' => $version, '@module' => $module)) . ' ' . $log_message;
```

It's OK to concatenate `$log_message` since it is a separate sentence and has already been translated.